native/src/main/java/io/ballerina/stdlib/persist/googlesheets/datastore/GoogleSheetsProcessor.java

```diff
@@ -0,0 +1,198 @@
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.ErrorType;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.StreamType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BStream;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.api.values.BTypedesc;
+import io.ballerina.stdlib.persist.googlesheets.Constants;
+import io.ballerina.stdlib.persist.googlesheets.ModuleUtils;
+
+import static io.ballerina.stdlib.persist.googlesheets.Constants.ERROR;
+import static io.ballerina.stdlib.persist.googlesheets.Constants.KEY_FIELDS;
+import static io.ballerina.stdlib.persist.googlesheets.Utils.getEntity;
+import static io.ballerina.stdlib.persist.googlesheets.Utils.getEntityFromStreamMethod;
+import static io.ballerina.stdlib.persist.googlesheets.Utils.getFieldTypes;
+import static io.ballerina.stdlib.persist.googlesheets.Utils.getKey;
+import static io.ballerina.stdlib.persist.googlesheets.Utils.getMetadata;
+import static io.ballerina.stdlib.persist.googlesheets.Utils.getPersistClient;
+import static io.ballerina.stdlib.persist.googlesheets.Utils.getRecordTypeWithKeyFields;
+
+/**
+ * This class provides the GoogleSheets query processing implementations for persistence.
+ *
+ * @since 0.3.0
+ */
+
+public class GoogleSheetsProcessor {
+
+    private GoogleSheetsProcessor() {};
+
+    public static BStream query(Environment env, BObject client, BTypedesc targetType) {
+        BString entity = getEntity(env);
+        BObject persistClient = getPersistClient(client, entity);
+        BArray keyFields = (BArray) persistClient.get(KEY_FIELDS);
+        RecordType recordType = (RecordType) targetType.getDescribingType();
+
+        RecordType recordTypeWithIdFields = getRecordTypeWithKeyFields(keyFields, recordType);
+        BTypedesc targetTypeWithIdFields = ValueCreator.createTypedescValue(recordTypeWithIdFields);
+        StreamType streamTypeWithIdFields = TypeCreator.createStreamType(recordTypeWithIdFields,
+                PredefinedTypes.TYPE_NULL);
+
+        BArray[] metadata = getMetadata(recordType);
+        BArray fields = metadata[0];
+        BArray includes = metadata[1];
+        BArray typeDescriptions = metadata[2];
+        BMap<BString, Object> typeMap = getFieldTypes(recordType);
+
+        Future balFuture = env.markAsync();
+        env.getRuntime().invokeMethodAsyncSequentially(
+                persistClient, Constants.RUN_READ_QUERY_METHOD,
+                null, null, new Callback() {
+                    @Override
+                    public void notifySuccess(Object o) {
+                        BStream sqlStream = (BStream) o;
```

Comment:
```suggestion
                        BStream stream = (BStream) o;
```

Reply:
fixed in [caecb93](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/1/commits/caecb93da85c1d1626d09ea912843b8249d12237)